source/codegen/templates/service_helpers.mako

```diff
@@ -208,8 +208,8 @@ ${initialize_standard_input_param(function_name, parameter)}\
   c_type_pointer = c_type.replace('[]','*')
   c_type_underlying_type = common_helpers.get_underlying_type_name(c_type)
 %>\
-% if c_type in ['ViConstString', 'const char*']:
-      ${c_type} ${parameter_name} = ${request_snippet}.c_str();\
+% if c_type in ['ViConstString', 'const char[]']:
+      auto ${parameter_name} = ${request_snippet}.c_str();\
```

Comment:
Note: A lot of the diffs come from this change so that we don't have to conditionally encode the destination type on the let-hand side (instead the type conversion is happening on passing the parameter).